src/util/sss_ssh.c

```diff
@@ -18,13 +18,53 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#define _GNU_SOURCE
+
+#include <string.h>
 #include <arpa/inet.h>
 
 #include "db/sysdb.h"
 #include "util/util.h"
 #include "util/crypto/sss_crypto.h"
 #include "util/sss_ssh.h"
 
+
+/* Check if the key_line we received from the backend includes the hostname or
+ * it is just the keytype and the key.
+ *
+ * Key lines have this format:
+ * marker (optional), hostnames, keytype, base64-encoded key, comment (optional)
+ *
+ * This is a very simplistic method based on looking for the provided hostname
+ * into the provided keyline, at the right position (the hostname could also
+ * be present in the comment at the end).
+ */
+static bool
+sss_ssh_key_has_host_name(const char *key_line, const char *hostname)
+{
+    const char *current = key_line;
+    const char *end;
+
+    /* Skip spaces */
+    while (*current != '\0' && *current == ' ') {
```

Comment:
`*current != '\0'` condition is excessive.

Reply:
It is testing the end of the string. Normally it should not happen, but how can we be sure?

Reply:
`*current == ' '` implies `*current != '\0'`

Reply:
Assuming a correctly formed "key". A malformed key might not have the expected space.

But I agree with you in that this should never happen.

Reply:
> A malformed key might not have the expected space.

It doesn't matter.
The condition `*current == ' '` automatically implies `*current != '\0'`

It's just: (A && B), where B=>A, can be reduced to (B)

Reply:
Sorry, I though we were talking about the other "macro", the one looking for a space.
In this case it is redundant.